cpp/oneapi/dal/detail/system_parameters.hpp

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <string>
+
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/detail/system_parameters_impl.hpp"
+
+namespace oneapi::dal {
+
+namespace detail {
+
+/// Stores system-related parameters that affect the performance of the algorithms.
+/// Those parameters can differ from the `get_global_context().get_cpu_info()`.
+///
+/// `cpu_info` reports the parameters available in hardware, when `system_parameters`
+/// are the software-enabled parameters that can differ from `cpu_info`.
+class system_parameters : public base {
+public:
+    /// Creates a new default ``system_parameters`` instance.
```

Comment:
Use of double backticks here, where single backticks are used in other places in this file. Keep the usage consistent

```suggestion
    /// Creates a new default `system_parameters` instance.
```